CMakeLists.txt

```diff
@@ -204,7 +204,7 @@ endif()
 
 ############ Compiler compatibility
 
-if(WIN32)
```

Comment:
This change is actually incorrect.
Currently, I only want to support MSVC on windows (either microsoft clang implementation or VC++). As microsoft-clang is actually just a different toolset, it's still MSVC that is defined, so I still want to check for the version and exit if it's too old.